### PR TITLE
better ldap config, auto-create users, logo config

### PR DESCRIPTION
**Pull request by nemesis621**
A bunch of fixes to make timetracker work here in our office.

### CODE REVIEW

File: src/Netresearch/TimeTrackerBundle/Controller/BaseController.php [hunk 1]
```diff
@@ -189,7 +189,11 @@ protected function setLoggedIn($user, $setCookie = true)
 
             return $this->render(
                 'NetresearchTimeTrackerBundle:Default:login.html.twig',
-                array('locale' => 'en')
+                array(
+                    'locale' => 'en',
+                    'apptitle' => $this->container->getParameter('app_title'),
+
```
Comment (CybotTM) [inline src/Netresearch/TimeTrackerBundle/Controller/BaseController.php]: superfluous new line

File: src/Netresearch/TimeTrackerBundle/Controller/DefaultController.php [hunk 2]
```diff
@@ -79,13 +88,18 @@ public function loginAction()
...
 
             $client = new LdapClient();
+            $client->setLogger($this->get('logger'));
+
             $client->setHost($this->container->getParameter('ldap_host'))
                 ->setPort($this->container->getParameter('ldap_port'))
                 ->setReadUser($this->container->getParameter('ldap_readuser'))
-                ->setReadPass($this->container->getParameter('ldap_readuser'))
+                ->setReadPass($this->container->getParameter('ldap_readpass'))
+                ->setBaseDn($this->container->getParameter('ldap_basedn'))
```
Comment (CybotTM) [inline src/Netresearch/TimeTrackerBundle/Controller/DefaultController.php]: is setBaseDn() now called twice?

File: src/Netresearch/TimeTrackerBundle/Controller/DefaultController.php [hunk 3]
```diff
@@ -95,7 +109,8 @@ public function loginAction()
                 'login'     => false, 
                 'message'   => $this->get('translator')->trans($e->getMessage()),
                 'username'  => $username,
-                'locale'    => 'en'
+                'locale'    => 'en',
+                'apptitle' => $this->container->getParameter('app_title'),
```
Comment (CybotTM) [inline src/Netresearch/TimeTrackerBundle/Controller/DefaultController.php]: check indent depth

File: src/Netresearch/TimeTrackerBundle/Controller/DefaultController.php [hunk 4]
```diff
@@ -104,6 +119,21 @@ public function loginAction()
             ->getRepository('NetresearchTimeTrackerBundle:User')
             ->findOneByUsername($username);
 
+        if (!$user) {
+            // create new user if users.username doesn't exist for valid ldap-authentication
```
Comment (CybotTM) [inline src/Netresearch/TimeTrackerBundle/Controller/DefaultController.php]: please make this behavior switchable by config option

File: src/Netresearch/TimeTrackerBundle/Controller/StatusController.php [hunk 5]
```diff
@@ -45,6 +45,7 @@ public function pageAction()
         $status = $this->getStatus($userId);
         return $this->render('NetresearchTimeTrackerBundle:Default:status.html.twig', array(
             'loginClass' => ($status['loginStatus'] ? 'status_active' : 'status_inactive'),
+            'apptitle' => $this->container->getParameter('app_title'),
```
Comment (CybotTM) [inline src/Netresearch/TimeTrackerBundle/Controller/StatusController.php]: check indent depth

File: src/Netresearch/TimeTrackerBundle/Helper/LdapClient.php [hunk 6]
```diff
@@ -35,6 +35,11 @@ class LdapClient
 	protected $_baseDn = 'dc=netresearch,dc=nr';
 
     /**
+     * @var string Accountname-Field in LDAP.
+     */
+	protected $_userNameField = 'sAMAccountName';
```
Comment (CybotTM) [inline src/Netresearch/TimeTrackerBundle/Helper/LdapClient.php]: check indent depth

File: src/Netresearch/TimeTrackerBundle/Helper/LdapClient.php [hunk 7]
```diff
@@ -72,7 +87,7 @@ protected function _verifyUsername()
 
         /* @var $result Ldap\Collection */
         $result = $ldap->search(
-            '(sAMAccountName=' . $this->_userName . ')',
+            '('.$this->_userNameField.'=' . ldap_escape($this->_userName) . ')',
```
Comment (CybotTM) [inline src/Netresearch/TimeTrackerBundle/Helper/LdapClient.php]: missing spaces around (string concatination) operators

File: src/Netresearch/TimeTrackerBundle/Helper/LdapClient.php [hunk 8]
```diff
@@ -96,15 +111,17 @@ protected function _verifyPassword(array $ldapEntry)
...
             'password'  => $this->_userPass,
             'baseDn'    => $this->_baseDn,
             'port'      => $this->_port,
+            'useSsl'    => $this->_useSSL,
         ));
 
         try {
             $ldap->bind();
         } catch (Ldap\Exception\LdapException $e) {
+            $this->logger->addError($e->getMessage());
```
Comment (CybotTM) [inline src/Netresearch/TimeTrackerBundle/Helper/LdapClient.php]: add check if logger is set, or set null logger by default

File: src/Netresearch/TimeTrackerBundle/Helper/LdapClient.php [hunk 9]
```diff
@@ -223,6 +240,34 @@ public function setBaseDn($base_dn)
...
 
     /**
+     * Determines whether SSL will be used for LDAP-connection or not
+     *
+     * @param boolean $useSSL
+     * @return $this
+     */
+    public function setUseSSL($useSSL)
+    {
+        $this->_useSSL = !empty($useSSL) && $this->_useSSL !== 0;
```
Comment (CybotTM) [inline src/Netresearch/TimeTrackerBundle/Helper/LdapClient.php]: i am confused, why so complicated?
What does 0 mean if set for this->_useSSL?
You cannot set useSSL to false!?

File: src/Netresearch/TimeTrackerBundle/Resources/views/Default/index.html.twig [hunk 10]
```diff
@@ -2,10 +2,12 @@
 <html>
     <head>
         <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
-        <title>Netresearch TimeTracker</title>
+        <title>{{ apptitle }}</title>
         <link rel="stylesheet" href="{{ asset('bundles/netresearchtimetracker/js/ext-js/resources/css/ext-all-gray.css') }}" type="text/css">
         <link rel="stylesheet" href="{{ asset('bundles/netresearchtimetracker/css/timetracker.css') }}" type="text/css">
         <script type="text/javascript">
+            var globalConfig = eval("(" + '{{ globalConfig|raw }}' + ")");
```
Comment (CybotTM) [inline src/Netresearch/TimeTrackerBundle/Resources/views/Default/index.html.twig]: doesn't this exposes (sensitive) configuration options?

**Comment by CybotTM** [comment]
Hi,

thank you very much for your contribution.
can you please have a look at my comments?

**Comment by nemesis621** [comment]
@CybotTM this issue is fixed in the current version of mogic-le/timetracker master

**Pull request merged**